conversions/octal_to_decimal.py

```diff
@@ -9,26 +9,40 @@ def oct_to_decimal(oct_string: str) -> int:
     >>> oct_to_decimal("-45")
     -37
     >>> oct_to_decimal("2-0Fm")
+    Traceback (most recent call last):
+    ...
     ValueError: Non-octal value was passed to the function
     >>> oct_to_decimal("")
+    Traceback (most recent call last):
+    ...
     ValueError: Empty string value was passed to the function
     >>> oct_to_decimal("19")
+    Traceback (most recent call last):
+    ...
     ValueError: Non-octal value was passed to the function
     """
+    # Strip oct_string of whitespaces
     oct_string = str(oct_string).strip()
     if not oct_string:
-        raise ValueError("Empty string was passed to the function")
+        raise ValueError("Empty string value was passed to the function")
+
+    # Check if oct_string is a negative value
     is_negative = oct_string[0] == "-"
     if is_negative:
+        # Remove (-) from oct_string
         oct_string = oct_string[1:]
-    if not all(0 <= int(char) <= 7 for char in oct_string):
+
+    # check if oct_string is an octal value and convert
+    if oct_string.isdecimal() and all(0 <= int(char) <= 7 for char in oct_string):
```

Comment:
These are redundant tests.  The first one cannon pass if the second one does not pass.  The original formulation is more consise.

Reply:
Thanks for reviewing. In the original, int(char) in the iterable passed to the all() function returns error that is not handled by the ValueError raised when char is not an integer value. It's why I modified it so.

Reply:
A ValueError is still raised.  This is why I encourage doctests.  It is not necessary that we raise all exceptions ourselves.  It is only that an appropriate exception is raised when garbage data is received.
```python
>>> int('a')
Traceback (most recent call last):
  File "<string>", line 1, in <module>
ValueError: invalid literal for int() with base 10: 'a'
```

Reply:
Ok